modules/exploits/linux/http/hp_van_sdn_token.rb

```diff
@@ -0,0 +1,132 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  # server: grizzly/2.2.16
+  HttpFingerprint = {pattern: [/grizzly/]}
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'HP VAN SDN Controller Hardcoded Service Token RCE',
+      'Description'    => %q{
+        This module exploits a hardcoded service token and weak sudo permissions
+        in HP(E) VAN SDN Controller <= 2.7.18.0503 to execute a payload as root.
+
+        Using the service token, a ZIP file containing the payload is uploaded
+        to the server, where it is automatically unzipped by the software.
+
+        A temporary directory is created in the process, thereby leading to a
+        directory traversal that is exploited to chain sudo commands.
+
+        The chained sudo commands are then used to execute the payload as root.
+      },
+      'Author'         => [
+        'Matt Bergin', # Vulnerability discovery and proof of concept
```

Comment:
`exploit`. It's more than a PoC. It works as described and achieves an interactive shell.

Reply:
Agreed, changing! The Metasploit terminology tends toward saying "PoC" vs. "module," but the original exploit is no less of an exploit.

Reply:
Maybe we should change our terminology? What about `PoC exploit`?

Reply:
Versus a PoC of the bug being triggered.

Reply:
Nah, KoreLogic's exploit is definitely weaponized. Dropping `PoC` altogether.

Reply:
While a lot of module commentary uses `PoC`, I've traditionally used `exploit` where appropriate, unless the exploit was shit.

I vote for `exploit` in instances where a reliable (or at least easily-configurable) exploit was provided; and `PoC` in instances where a trigger or plain text non-executable PoC (such as a raw HTTP request) was provided. While there may be an area of grey between the two categories, I don't see that `PoC exploit` is any clearer.

Wikipedia's definition doesn't help my argument :

> An exploit (from the English verb to exploit, meaning "to use something to one’s own advantage") is a piece of software, a chunk of data, or a sequence of commands that takes advantage of a bug or vulnerability to cause unintended or unanticipated behavior to occur on computer software, hardware, or something electronic (usually computerized).

Reply:
I agree with your definitions and have traditionally used the same before I worked on Metasploit. I'll realign with the more correct wording. :bowing_man: